mne/proj.py

```diff
@@ -97,15 +97,20 @@ def _compute_proj(data, info, n_grad, n_mag, n_eeg, desc_prefix, verbose=None):
         if n == 0:
             continue
         data_ind = data[ind][:, ind]
-        U = linalg.svd(data_ind, full_matrices=False,
-                       overwrite_a=True)[0][:, :n]
-        for k, u in enumerate(U.T):
+        # data is the covariance matrix: U * S**2 * Ut
+        U, Sexp2, _ = linalg.svd(data_ind, full_matrices=False,
+                                 overwrite_a=True)
+        U = U[:, :n]
+        exp_var = Sexp2 / Sexp2.sum()
+        exp_var = exp_var[:n]
+        for k, temp in enumerate(zip(U.T, exp_var)):
```

Comment:
for k, temp in enumerate(zip(U.T, exp_var)):

->

for k, (u, var) in enumerate(zip(U.T, exp_var)):